app/uk/gov/hmrc/plasticpackagingtax/registration/services/EmailVerificationService.scala

```diff
@@ -0,0 +1,75 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.services
+
+import com.google.inject.Inject
+import play.api.mvc.AnyContent
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.plasticpackagingtax.registration.connectors.EmailVerificationConnector
+import uk.gov.hmrc.plasticpackagingtax.registration.models.emailverification._
+import uk.gov.hmrc.plasticpackagingtax.registration.models.request.JourneyRequest
+
+import javax.inject.Singleton
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class EmailVerificationService @Inject() (emailVerificationConnector: EmailVerificationConnector)(
+  implicit ec: ExecutionContext
+) {
+
+  def isEmailVerified(
+    email: String
+  )(implicit req: JourneyRequest[AnyContent], hc: HeaderCarrier): Future[Boolean] =
+    emailVerificationConnector.getStatus(req.user.credId).map {
+      case Right(resp) =>
+        resp.exists { emailStatuses =>
+          emailStatuses.emails.exists {
+            case es @ EmailStatus(_, true, _) if es.emailAddress == email => true
+            case _                                                        => false
+          }
+        }
+      case Left(ex) => throw ex
+    }
+
+  def sendVerificationCode(
+    email: String
+  )(implicit req: JourneyRequest[AnyContent], hc: HeaderCarrier): Future[String] =
+    emailVerificationConnector.create(
+      CreateEmailVerificationRequest(credId = req.user.credId,
+                                     continueUrl =
+                                       "/register-for-plastic-packaging-tax/amend-registration",
+                                     origin = "ppt",
+                                     accessibilityStatementUrl = "/accessibility",
+                                     email = Email(address = email, enterUrl = "/start"),
+                                     backUrl = "/back",
+                                     pageTitle = "PPT Title",
+                                     deskproServiceName = "plastic-packaging-tax"
+      )
+    ).map {
+      case Right(resp) => resp.split("/").slice(0, 4).last // TODO: can we make this less 'brittle'?
+      case Left(ex)    => throw ex
+    }
+
+  def checkVerificationCode(code: String, email: String, journeyId: String)(implicit
+    hc: HeaderCarrier
+  ): Future[EmailVerificationJourneyStatus.Value] =
+    emailVerificationConnector.verifyPasscode(journeyId, VerifyPasscodeRequest(code, email)).map {
+      case Right(verificationStatus) => verificationStatus
+      case Left(ex)                  => throw ex
+    }
+
```

Comment:
It probably makes sense to refactor the existing email verification controllers to make use of this new Service class.